code/game/machinery/camera/tracking.dm

```diff
@@ -18,25 +12,18 @@
 		if (tempnetwork.len)
 			T[text("[][]", C.c_tag, (C.can_use() ? null : " (Deactivated)"))] = C
 
-	track.cameras = T
 	return T
 
-
-/mob/living/silicon/ai/proc/ai_camera_list(camera)
-	if (!camera)
-		return 0
-
-	var/obj/machinery/camera/C = track.cameras[camera]
-	src.eyeobj.setLoc(C)
-
-	return
+/mob/living/silicon/ai/proc/show_camera_list()
+	var/list/cameras = get_camera_list()
+	var/camera = input(src, "Choose which camera you want to view", "Cameras") as null|anything in cameras
+	switchCamera(cameras[camera])

```

Comment:
this needs a check if the user actually selected a camera

Reply:
That's handled by the QDELETED check in switchCamera

Reply:
On second thought, the QDELETED check does need moved slightly.

Reply:
This is fine actually, I was mainly worried that it would runtime on `cameras[null]` but apparently byond simply doesn't care.